slackbot/dispatcher.py

```diff
@@ -138,9 +138,14 @@ def loop(self):
         while True:
             events = self._client.rtm_read()
             for event in events:
-                if event.get('type') != 'message':
-                    continue
-                self._on_new_message(event)
+                if event.get('type') == 'message':
```

Comment:
More about style than performance. Even on a very busy slack, neither memory nor cpu is going to be a factor with this code.